java/vector/src/main/java/org/apache/arrow/vector/complex/NonNullableStructVector.java

```diff
@@ -303,7 +303,7 @@ public boolean equals(int index, ValueVector to, int toIndex) {
     if (to == null) {
```

Comment:
StructVector extends from this one. it should compare the validity bits too.

Reply:
Thanks, fixed.